Build failures, Arch Linux and GCC 15.

Failure in openssl:
```
/usr/include/c++/15.1.1/ciso646:46:4: error: #warning "<ciso646> is deprecated in C++17, use <version> to detect implementation-specific macros" [-Werror=cpp]
   46 | #  warning "<ciso646> is deprecated in C++17, use <version> to detect implementation-specific macros"
```

This can be resolved by adding `sed -i 's|-Werror||' '../CMakeLists.txt'` in the `$(boringssl_static_libs):` section of the makefile. This happens on tests. Later versions of openssl support the BUILD_TESTING variable to turn off building tests.

Next failure:
```
configure: error: OpenSSL libs and/or directories were not found where specified!
```

Curl fails to configure.

--- a/patches/boringssl.patch
+++ b/patches/boringssl.patch
@@ -1376,3 +1376,16 @@ index 6d46f886b..005b30f17 100644
 +  return set_sigalg_prefs(&ctx->delegated_credentials,
 +                          MakeConstSpan(sigalgs.data(), sigalgs.size()));
 +}
+diff --git a/third_party/googletest/googletest/include/gtest/internal/gtest-port.h b/third_party/googletest/googletest/include/gtest/internal/gtest-port.h
+index 4cfc16c03..0bbc208c0 100644
+--- a/third_party/googletest/googletest/include/gtest/internal/gtest-port.h
++++ b/third_party/googletest/googletest/include/gtest/internal/gtest-port.h
+@@ -282,7 +282,7 @@
+ 
+ // Detect C++ feature test macros as gracefully as possible.
+ // MSVC >= 19.15, Clang >= 3.4.1, and GCC >= 4.1.2 support feature test macros.
+-#if GTEST_INTERNAL_CPLUSPLUS_LANG >= 202002L && \
++#if GTEST_INTERNAL_CPLUSPLUS_LANG >= 201703L && \
+     (!defined(__has_include) || GTEST_INTERNAL_HAS_INCLUDE(<version>))
+ #include <version>  // C++20 and later
+ #elif (!defined(__has_include) || GTEST_INTERNAL_HAS_INCLUDE(<ciso646>))

